### ISSUE TITLE
[expr.const] Consider using "non-constant library call" more generally

### ISSUE THREAD

**Issue by frederick-vs-ja** [cplusplus/draft#5813]
Currently serveral bullets in [expr.const] are mentioning standard library components.

Constant evaluation failure required:
> (5.28) an invocation of the `va_arg` macro ([cstdarg.syn]);
> (5.29) a non-constant library call ([defns.nonconst.libcall]); or

Unspecified whether constant evaluation fails:
> (5.31) an operation that has undefined behavior as specified in [library] through [thread],
> (5.32) an invocation of the `va_start` macro ([cstdarg.syn]), or

And in [bit.pow.two]/8, constant evaluation failure is required when `bit_ceil` call results in UB (generally not required for library UB).
> _Remarks_: A function call expression that violates the precondition in the _Preconditions_: element is not a core constant expression ([expr.const]).

It may be a consistency improvement to specify all of these cases with "non-constant library call". Although only function invocation is specified in "non-constant library call", it seems easy to extend the definition to cover function-like macros.

**Comment by frederick-vs-ja** [comment]
[P2996](https://wg21.link/p2996) is introducing _Constant When_.

**Comment by jwakely** [comment]
> [P2996](https://wg21.link/p2996) is introducing _Constant When_.

Yes, but I think it's expected to be replaced with something else. We should not start basing anything else on "Constant when".

**Comment by frederick-vs-ja** [comment]
> > [P2996](https://wg21.link/p2996) is introducing _Constant When_.
> 
> Yes, but I think it's expected to be replaced with something else. We should not start basing anything else on "Constant when".

Is this still the case? Most _Constant When_ are replaced with _Throws_, but the rest look appropriate to me.
